src/lean_spec/subspecs/xmss/aggregation.py

```diff
@@ -0,0 +1,132 @@
+"""
+lean-multisig aggregation helpers bridging leanSpec containers to native bindings.
+
+This module wraps the Python bindings exposed by the `lean-multisig` project to provide
+XMSS signature aggregation + verification.
+
+The aggregated signatures are stored as raw payload bytes produced by
+`lean_multisig.aggregate_signatures`.
+"""
+
+from __future__ import annotations
+
+from functools import lru_cache
+from typing import Sequence
+
+from lean_spec.subspecs.xmss.containers import PublicKey, Signature
+from lean_spec.types import Uint64
+
+
+class LeanMultisigError(RuntimeError):
+    """Base exception for lean-multisig aggregation helpers."""
+
+
+class LeanMultisigUnavailableError(LeanMultisigError):
+    """Raised when the lean-multisig Python bindings cannot be imported."""
+
+
+class LeanMultisigAggregationError(LeanMultisigError):
+    """Raised when lean-multisig fails to aggregate or verify signatures."""
+
+
+@lru_cache(maxsize=1)
+def _import_lean_multisig():
+    try:
+        import lean_multisig  # type: ignore
+    except ModuleNotFoundError as exc:  # pragma: no cover - import is environment-specific
+        raise LeanMultisigUnavailableError(
+            "lean-multisig bindings are required. Install them with `uv pip install lean-multisig` "
+            "(or your local editable install) from the leanSpec repository."
+        ) from exc
+    return lean_multisig
+
+
+@lru_cache(maxsize=1)
+def _ensure_prover_setup() -> None:
+    """Run the (expensive) prover setup routine exactly once."""
+    _import_lean_multisig().setup_prover()
+
+
+@lru_cache(maxsize=1)
+def _ensure_verifier_setup() -> None:
+    """Run the verifier setup routine exactly once."""
+    _import_lean_multisig().setup_verifier()
+
+
+def _coerce_epoch(epoch: int | Uint64) -> int:
+    value = int(epoch)
+    if value < 0 or value >= 2**32:
+        raise ValueError("epoch must fit in uint32 for lean-multisig aggregation")
+    return value
+
+
+def aggregate_signatures(
+    public_keys: Sequence[PublicKey],
+    signatures: Sequence[Signature],
```

Comment:
this is XmssSignature sequene? I don't think we have the recusive signature signing as of now?